mapping/react-leaflet.md

```diff
@@ -0,0 +1,433 @@
+
+This guide also shows one of several ways to compose reusable components in a React app. Hopefully, if this guide succeeds at that, then you should be able to take these lessons and apply them to your project, i.e. organize your components in a way that works for your project or makes more sense to your brain.
+
+Before you begin, you should read [the introduction section for `react-leaflet`](https://react-leaflet.js.org/docs/start-introduction), but the rest of their setup guide is lacking. This guide hopes to help others avoid that pain.
+
+## Setup your React app
+Assuming you have npm installed:
+```bash
+npx create-react-app hello-leaflet
+cd hello-leaflet
+```
+
+Install `leaflet@1.7.1`, `react-leaflet@3.1.0`, and `@react-leaflet/core@1.0.2`:
+```bash
+npm install leaflet@1.7.1 react-leaflet@3.1.0 @react-leaflet/core@1.0.2
+```
+These versions are important due to an incompatibility with `react-leaflet/core` and `create-react-app`, at the time of writing. For more info about that, see this [stackoverflow with a variety of workarounds](https://stackoverflow.com/questions/67552020/how-to-fix-error-failed-to-compile-node-modules-react-leaflet-core-esm-pat), and this [Github issue about the bug](https://github.com/PaulLeCam/react-leaflet/issues/877).
+
+
+Test that your React app works:
+```bash
+npm start
+```
+
+## Create a Base Map
+In `src/maps/base.js`, create this component:
+```jsx
+import React from 'react'
+import { MapContainer, TileLayer } from 'react-leaflet'
+import 'leaflet/dist/leaflet.css'
+
+function BaseMap({ center, zoom, className, children }) {
+    return (
+      <MapContainer
+        center={center} 
+        zoom={zoom}
+        className={className}>
+          <TileLayer
+            url="https://{s}.basemaps.cartocdn.com/light_nolabels/{z}/{x}/{y}{r}.png"
+            attribution='&copy; <a href="https://www.openstreetmap.org/copyright" target="_parent">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions" target="_parent">CARTO</a>'
+          />
+          {children}
+      </MapContainer>
+    )
+}
+
+export default BaseMap
+```
+
+This is a reusable component for making maps. Whenever you use it, you can pass props into it, like `center`, `zoom`, `className`, and `children`. The `children` argument enables you to build a map with custom layers. This is a powerful feature of React, and helps you "compose" resuable components in a way that makes sense for your UI. You will see how this works when you add the GeoJSON layer.
+
+**It's very important that you import `leaflet/dist/leaflet.css` to your `BaseMap`. This isn't obvious from the React Leaflet documentation.**
```

Comment:
What do you think about including a description of what happens if you don't include the styles, to help the reader troubleshoot?